src/Sylius/Bundle/ApiBundle/Provider/CustomerProvider.php

```diff
@@ -30,17 +33,33 @@ final class CustomerProvider implements CustomerProviderInterface
     /** @var CustomerRepositoryInterface */
     private $customerRepository;
 
+    /** @var UserContextInterface */
+    private $userContext;
+
     public function __construct(
         CanonicalizerInterface $canonicalizer,
         FactoryInterface $customerFactory,
-        CustomerRepositoryInterface $customerRepository
+        CustomerRepositoryInterface $customerRepository,
+        UserContextInterface $userContext
     ) {
         $this->canonicalizer = $canonicalizer;
         $this->customerFactory = $customerFactory;
         $this->customerRepository = $customerRepository;
+        $this->userContext = $userContext;
+    }
+
+    public function get(): ?CustomerInterface
```

Comment:
I am not sure about this function, firstly this service becomes stateful, secondly, you are mixing two logic (or 3): getting the customer from `userContext`, and getting customer from database (or creating it).